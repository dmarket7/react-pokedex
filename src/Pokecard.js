import React from 'react';

class Pokecard extends React.Component {
  render() {
    const pokeName = this.props.name;
    const imageURL = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${this.props.id}.png`
    const type = this.props.type;
    const experience = this.props.experience;

    return (
      <div>
        <div className={`card`} style={{width: `18rem`}}>
          <h3 className={`card-title`}>{ pokeName }</h3>
          <img src={imageURL} className={`card-img-top`} alt={"Pokecard"}></img>
          <small>{type}</small>
          <small>{experience}</small>
        </div>
      </div>
    );
  };
}

export default Pokecard;