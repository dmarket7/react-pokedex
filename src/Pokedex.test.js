import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Pokecard
  from "./Pokecard";

it("renders using shallow", function() {
  // this test passes
  shallow(<Pokecard />);
});

it("renders using mount", function() {
  // this test fails!
  mount(<Pokecard />);
});

it("matches snapshot for adopted dogs", function() {
  let wrapper = shallow(<Pokecard name="Eevee" id={4} type="fire" experience={62} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

