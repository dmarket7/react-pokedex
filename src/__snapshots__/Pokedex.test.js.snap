// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for adopted dogs 1`] = `
<div>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <h3
      className="card-title"
    >
      Eevee
    </h3>
    <img
      alt="Pokecard"
      className="card-img-top"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
    />
    <small>
      fire
    </small>
    <small>
      62
    </small>
  </div>
</div>
`;
